File: app/src/main/java/com/boardgamegeek/provider/GamesIdProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.GamePolls;
import com.boardgamegeek.provider.BggContract.Games;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesIdProvider extends BaseProvider {
	private final GamesProvider provider = new GamesProvider();

	@Override
	protected SelectionBuilder buildExpandedSelection(Uri uri) {
		int gameId = Games.getGameId(uri);
		String GAME_POLLS_COUNT = "(SELECT COUNT(" + GamePolls.POLL_NAME + ") FROM " + Tables.GAME_POLLS
			+ " WHERE " + Tables.GAME_POLLS + "." + GamePolls.GAME_ID + "=" + Tables.GAMES + "." + Games.GAME_ID + ")";
		return new SelectionBuilder()
			.table(Tables.GAMES)
			.map(Games.POLLS_COUNT, GAME_POLLS_COUNT)
			.whereEquals(Tables.GAMES + "." + Games.GAME_ID, gameId);
	}

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		int gameId = Games.getGameId(uri);
		return provider.buildSimpleSelection(uri).whereEquals(Games.GAME_ID, gameId);
	}

	@Override
	protected String getPath() {
		return addIdToPath(provider.getPath());
	}

	@Override
	protected String getType(Uri uri) {
		return Games.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesIdProvider.kt
package com.boardgamegeek.provider

import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteQueryBuilder
import android.net.Uri
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggContract.Games
import com.boardgamegeek.provider.BggDatabase.Tables

class GamesIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = Games.CONTENT_ITEM_TYPE

    override val path = "$PATH_GAMES/#"

    override fun query(
        db: SQLiteDatabase,
        uri: Uri, // content://com.boardgamegeek/games/13
        projection: Array<String>?,
        selection: String?,
        selectionArgs: Array<String>?,
        sortOrder: String?
    ): Cursor? {
        val gameId = Games.getGameId(uri)
        val qb = SQLiteQueryBuilder()
        qb.tables = Tables.GAMES
        qb.appendWhere("${Games.Columns.GAME_ID} = $gameId")
        return qb.query(db, projection, selection, selectionArgs, null, null, sortOrder)
    }
}
